enhancements/sig-architecture/123-addon-multiple-configs/README.md

```diff
@@ -0,0 +1,230 @@
+which will override the config in place `1`. The final effective addon configs are listed in the mca.status.
+
+Now we are seeing the requirement to set multiple configs for the same GVK. For example, for `OpenTelemetryCollector`, one CR might be configured to 
+collect traces and another to collect logs on the same cluster. This proposal aims to satisfy that requirement.
+
+### Goals
+
+- Enhance `ClusterManagementAddon` and `ManagementClusterAddon` API to allow users to configure multiple addon configs with the same GVK.
+
+### Non-Goals
+
+- `ClusterManagementAddon` `spec.supportedConfigs` does not allow user to configure mutiple addon configs with same GVK in the `defaultConfig`.
+
+## Proposal
+
+### User Stories
+
+1. The admin can configure the `ClusterManagementAddon` API to set mutilple configs with same gvk per install strategy.
+2. The admin can configure the `ManagementClusterAddon` API to set mutilple configs with same gvk for specific cluster.
+
+## Design Details
+
+### API changes
+
+`cma.spec.installStrategy.placements[].configs` and `mca.spec.configs` already support mutiple configs. 
+
+```go
+	// Configs is the configuration of managedClusterAddon during installation.
+	// User can override the configuration by updating the managedClusterAddon directly.
+	// +optional
+	Configs []AddOnConfig `json:"configs,omitempty"`
+```
+
+### Effective addon configs
+
+As the motivation mentioned, today there are 3 places to set addon configurations. 
+
+Currently, since each GVK supports only one config, the GVK is used as the identifier for configs. When configs with same GVK configured in multiple places, it follows the override sequence and 
+only one config is listed as the effective config in the mca.status.
+
+With the support for multiple configs with the same GVK, `GVK + namespace + name` becomes the identifier. 
+We need to reconsider the logic for setting mca.status when configuring multiple configs in multiple places.
+
+We present three options using the example below (omitting the group and namespace for simplicity). 
+We do not expect to support all but want to start a discussion and eventually reach a consensus on which option should be adopted by the community.
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: example
+spec:
+  supportedConfigs:
+    - resource: r1
+      defaultConfig:
+        name: n1
+    - resource: r2
+  installStrategy:
+    type: Placements
+    placements:
+      - name: example
+        namespace: example
+        configs:
+          - resource: r2
+            name: n1
+          - resource: r2
+            name: n2
+
+
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: example
+  namespace: cluster1
+spec:
+  installNamespace: open-cluster-management-agent-addon
+  configs:
+    - resource: r2
+      name: n3
+```
+
+#### Merge
```

Comment:
For this strategy: what will happen when we have one resource in the `ManagedClusterAddon` and another one with the same name, GVK and namespace in `ClusterManagemenetAddon`? I would expect an automatic override but I'm not sure if you want to do something to keep both of them in order or something.

Reply:
if the one in `ClusterManagemenetAddon` has the same gvk, name and namespace as the one in the `ManagedClusterAddon`, they are actually the same config. In this case, since merge uses gvk + name + namespace as the key, so only one will be listed in the `mca.status`.